### DISCUSSION TITLE
[VTS] Feedback from Community

### DISCUSSION THREAD

**Discussion by onmax** [nimiq/developer-center#24] [General]
# Liveness and slot allocation

- Liveness is seen as a measure of luck. Slot allocation appears to be random, but is influenced by the amount of NIM staked.
- Slots are allocated based on the NIM deposited in the staking contract, from own or delegated stakes.

# Geographical distribution

- The geographical location of validators is important to avoid centralisation and potential network failure if a single provider (e.g. Hetzner) bans the network.
- Incentivising geographic distribution could prevent centralisation, but could be difficult to verify due to potential spoofing.

# Incentives and validator performance

- Validators and history node hosts could be incentivised through specific mechanisms, although verification methods are currently unclear.
- Users prioritise high APY (Annual Percentage Yield) when choosing validators. The network should focus on GEO distribution and stakes concentration.
- Validators that do not pay delegates should be penalised, as non-payment affects trust and APY calculations.

# Trust and reliability

- A trust score may not be fair if the start of the network is unstable, affecting the reliability scores of early validators.
- The implementation of the trust score should take into account the stability of the network.

# APY and financial considerations

- The APY for users may decrease if validators do not pay out, but validators are still rewarded.
- Financial incentives to operate validators in less populated or more expensive areas, such as Asia, could be beneficial.

# Tunneling and latency

- Tunneling can be used to spoof geographic locations, but may not be practical due to latency issues.
Starlink offers low latency, but this may not be sufficient for strict geographical requirements.
- Dedicated routes and tunneling to different regions can affect latency, but this may not always be cost effective.

**Comment by maestroi** [comment]
I think software reliability plays into this as well, cause currently it can lose connection to network and never recover which is a penalty, but hard for operator to "fix"

**Comment by joeleol** [comment]
An additional comment regarding the potential measurement of paying-out to stakers: I will be imposing a minimum stake of approximately 5-10K NIM in my pool, therefore anyone that stakes less than 5-10K NIM will not get paid-out - as I cannot in any way block these stakers.

**Comment by Albermonte** [comment]
> A trust score may not be fair if the start of the network is unstable, affecting the reliability scores of early validators.


From the [docs](https://deploy-preview-21--nimiq-developer-center.netlify.app/learn/validator-trust-score#the-vts-algorithm), we are using the constant `m` that is the same for every validator. This means that older validators will have higher liveness score than newer validator, even if the network is unstable.

Worst case would be if the network is so unstable that causes that a validator misses all the slots assigned, making the reliability score to be close to 0, but their liveness score will be much higher than any new validator. So even in this case, an older validator should have an advantage (assuming that there are no more network issues and both validators can produce blocks normally)
